docs/source/tutorials/openstack.rst

```diff
@@ -0,0 +1,229 @@
+
+.. code:: bash
+
+	$ linchpin fetch --root docs/source/examples/workspaces openshift-on-openstack --branch 1.7.2 --dest ./fetch-example https://github.com/CentOS-PaaS-SIG/linchpin
+
+You can even choose to fetch only a certain component of the workspace.  For example, if you only wish to fetch the topologies you can add :code:`--type topologies`.  If you were able to fetch a complete workspace, you can skip to `Up`_
+
+Initialization
+--------------
+
+Assuming you are creating a workspace from scratch, you can run :code:`linchpin init` to initialize a workspace.  The following line of code will create a linchpin.conf, dummy PinFile, and README.rst in a directory called "simple"
+
+.. code:: bash
+
+	$ linchpin init simple
+
+The PinFile contains a single target, called simple, which contains a topology but no layout.  A group of related provisioning tasks is called a target.  Each target has a topology, which can contain many resource groups, and an optional layout.  We'll explain what each of those means later on in further detail
+
+Creating a Topology
+-------------------
+
+Now that we have a PinFile, its time to add the code for an OpenStack server.  Edit your PinFile so it looks like the one below.
+
+.. code:: yaml
+
+	simple:
+	  topology:
+		topology_name: simple
+		resource_groups:
+		  - resource_group_name: os_simple
+			resource_group_type: openstack
+			resource_definitions:
+			  - name: simple_keypair
+				role: os_keypair
+			  - name: simple_server
+				role: os_server
+				flavor: m1.small
+				keypair: simple_keypair
+				count: 1
+
+There are a number of other fields available for these two roles.  Information about those fields as well as the other OpenStack roles can be found on the `OpenStack provider page`_.
+
+A :term:`resource group` is a group of resources related to a single provider.  In this example we have an openstack resource group that defines two different types of openstack resources.  We could also define an AWS resource group below it that provisions a handful of EC2 nodes.  A single resource group can contain many :term:`resource definitions`. A resource definition details the requirements for a specific resource.  Multiple resources can be provisioned from a single resource definition by editing the :code:`count` field, but all non-unique properties of the resources will be identical.
+
+.. _openstack provider page: ../openstack.rst
+
+
+
+Creating a Layout
+-----------------
+
+LinchPin can use layouts to describe what an Ansible inventory might look like after provisioning.  Layouts can include information such as IP addresses, zones, and FQDNs.  Under the simple key, put the following data:
+
+.. code:: yaml
+
+	layout:
```

Comment:
This should be indented the same as topology above right?  it looks like it should be pushed over a little more?

Reply:
Yes, I'll fix that